File: practice/algorithms/strings/funny_string/FunnyString.js
function processData(input) {
    var lines = input.split('\n');
    for(var i = 1; i < lines.length; i++) {
        if(isFunny(lines[i])) {
            console.log("Funny");
        } else {
            console.log("Not Funny");
        }
    }
}

function isFunny(s) {
    for(var j = 0; j < s.length - 1; j++) {
        if(Math.abs(s.charCodeAt(j + 1) - s.charCodeAt(j)) !==
            Math.abs(s.charCodeAt(s.length - j - 2) - s.charCodeAt(s.length - j - 1))) {
            return false;
        }
    }
    return true;
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});

File: practice/algorithms/strings/funny_string/FunnyString.java
import java.util.Scanner;

public class FunnyString 
{

	public static void main(String[] args)
	{
		Scanner stdin = new Scanner(System.in);
		int tests = Integer.parseInt(stdin.nextLine());
		for(int i = 0; i < tests; i++)
		{
			String s = stdin.nextLine();
			if(isFunny(s))
			{
				System.out.println("Funny");
			}
			else
			{
				System.out.println("Not Funny");
			}
			
		}
		stdin.close();
	}
	
	private static boolean isFunny(String s)
	{
		for(int j = 0; j < s.length() - 1; j++)
		{
			if(Math.abs(Character.codePointAt(s, j + 1) - Character.codePointAt(s, j)) !=
					Math.abs(Character.codePointAt(s, s.length() - j - 2) - Character.codePointAt(s, s.length() - j - 1)))
			{
				return false;
			}
		}
		return true;
	}
}
